libminifi/src/utils/BackTrace.cpp

```diff
@@ -42,10 +42,9 @@ namespace {
    */
   optional<std::string> demangle_symbol(const char* symbol_name) {
     int status;
-    char* demangled = abi::__cxa_demangle(symbol_name, nullptr, nullptr, &status);
+    std::unique_ptr<char> demangled(abi::__cxa_demangle(symbol_name, nullptr, nullptr, &status));
```

Comment:
`free` should be used to delete this result: https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html

Reply:
Updated in 56ad23b1bafb764f8bf4e6e66b71af3641fea87d

Reply:
Couldnt we use std::unique_ptr<char, FreeDeleter> instead?

Reply:
That would also fix the exception safety issue here. (What if the `std::string` allocation throws.)

Reply:
Good idea, updated in c7c57deb66f14433990fd21f6f1fe8ad7184d117